### PR TITLE
fix(core): tsconfig.base.json module setting should not break local plugins

### PR DESCRIPTION
**Pull request by AgentEnder**
## Current Behavior
`tsconfig.base.json` influences local plugin runtime as it is read for all compiler options which are then fed into ts-node or swc-node. 

When setting module to `ESNEXT` or similar, this can break plugin usage.

## Expected Behavior
Plugins run in a known environment. They will always run under node or a similar runtime, so we can provide an appropriate tsconfig. Only the paths from tsconfig.base.json should really matter.

## Related Issue(s)

There's a community slack thread here: https://nrwlcommunity.slack.com/archives/CMFKWPU6Q/p1674653626198069

### CODE REVIEW

File: docs/shared/packages/nx-plugin/nx-plugin.md [hunk 1]
```diff
@@ -155,3 +155,9 @@ To make sure that assets are copied to the dist folder, open the plugin's `proje
 ## Using your Nx Plugin
 
 To use your plugin, simply list it in `nx.json` or use its generators and executors as you would for any other plugin. This could look like `nx g @my-org/my-plugin:lib` for generators or `"executor": "@my-org/my-plugin:build"` for executors. It should be usable in all of the same ways as published plugins in your local workspace immediately after generating it. This includes setting it up as the default collection in `nx.json`, which would allow you to run `nx g lib` and hit your plugin's generator.
+
+{% callout type="warning" title="string" %}
+
+Nx uses the paths from tsconfig.base.json when running plugins locally, but uses the recommended tsconfig for node 16 for other compiler options. See https://github.com/tsconfig/bases/blob/main/bases/node16.json
```
Comment (bcabanes) [inline docs/shared/packages/nx-plugin/nx-plugin.md:161]:
```suggestion
Nx uses the paths from `tsconfig.base.json` when running plugins locally, but uses the recommended tsconfig for node 16 for other compiler options. See https://github.com/tsconfig/bases/blob/main/bases/node16.json
```

File: docs/shared/recipes/plugins/local-executors.md [hunk 2]
```diff
@@ -134,6 +134,12 @@ Options: {
 Hello World
 ```
 
+{% callout type="warning" title="string" %}
+
+Nx uses the paths from tsconfig.base.json when running plugins locally, but uses the recommended tsconfig for node 16 for other compiler options. See https://github.com/tsconfig/bases/blob/main/bases/node16.json
```
Comment (bcabanes) [inline docs/shared/recipes/plugins/local-executors.md:139]:
```suggestion
Nx uses the paths from `tsconfig.base.json` when running plugins locally, but uses the recommended tsconfig for node 16 for other compiler options. See https://github.com/tsconfig/bases/blob/main/bases/node16.json
```

File: packages/nx/src/utils/nx-plugin.ts [hunk 3]
```diff
@@ -22,6 +22,13 @@ import {
   findProjectForPath,
 } from '../project-graph/utils/find-project-for-path';
 import { normalizePath } from './path';
+import { join } from 'path';
+
+import type {
```
Comment (FrozenPandaz) [inline packages/nx/src/utils/nx-plugin.ts:27]: Remove this

**Review by FrozenPandaz — APPROVED** [review]

**Review by bcabanes — APPROVED** [review]

**Comment by isaacplmann** [comment]
Throwing callouts in 4 different docs isn't the right way to document this, but I don't have the headspace right now to think of a better solution.  I'll look at it again tomorrow morning.

**Comment by AgentEnder** [comment]
> Throwing callouts in 4 different docs isn't the right way to document this, but I don't have the headspace right now to think of a better solution. I'll look at it again tomorrow morning.

We can revisit how to document it at some point, for now I think the callouts serve their purpose.

**Pull request merged**